hasher-matcher-actioner/hmalib/lambdas/api/middleware.py

```diff
@@ -63,9 +64,24 @@ def from_dict(cls, d: t.Dict) -> "DictParseable":
         raise NotImplementedError
 
 
+class DictParseableWithSignalTypeMapping:
+    """
+    A special case of dict parseables where the dynamic signal type config is
+    required.
```

Comment:
During rewriting the CLI I constantly struggled on whether the list of signal types should be stored statically or not (SignalType.get_all(), which would error if not set in init). The number of places where you want to treat it as a statically known thing is high.

Reply:
Even though it is painful in the short run to get signal_type_mapping over to _everything_ under the sun, I believe it lets us truly achieve your vision of "if it works in pytx, it will work in hma". 

To that end, I don't mind the requirement to have signal_type_mapping _everywhere_.